frontend/packages/console-app/src/components/modals/resource-limits/__tests__/ResourceLimitsModal.spec.tsx

```diff
@@ -52,15 +57,28 @@ describe('ResourceLimitsModal Form', () => {
         },
       },
     } as Props;
-    ResourceLimitsModalWrapper = shallow(<ResourceLimitsModal {...formProps} />);
   });
 
-  it('should render ResouceLimitSection', () => {
-    expect(ResourceLimitsModalWrapper.find(ResourceLimitSection)).toHaveLength(1);
+  it('renders the modal with the correct title and initial elements', () => {
+    renderWithProviders(<ResourceLimitsModal {...formProps} />);
+
+    expect(screen.getByText('Edit resource limits')).toBeVisible();
+    expect(screen.getByRole('form', { name: 'Resource limits modal' })).toBeVisible();
+    expect(screen.getByRole('button', { name: 'Cancel' })).toBeVisible();
+    expect(screen.getByRole('button', { name: 'Save' })).toBeVisible();
+  });
+
+  it('calls the cancel function when the Cancel button is clicked', async () => {
+    renderWithProviders(<ResourceLimitsModal {...formProps} />);
+
+    await fireEvent.click(screen.getByRole('button', { name: 'Cancel' }));
+    expect(formProps.cancel).toHaveBeenCalledTimes(1);
   });
 
-  it('should call handleSubmit on form submit', () => {
-    ResourceLimitsModalWrapper.simulate('submit');
+  it('calls the handleSubmit function when the form is submitted', async () => {
+    renderWithProviders(<ResourceLimitsModal {...formProps} />);
+
+    await fireEvent.submit(screen.getByRole('form', { name: 'Resource limits modal' }));
     expect(formProps.handleSubmit).toHaveBeenCalledTimes(1);
```

Comment:
Shall we add some checkpoints covering some negative tests? 
For example 
- check `Save` button is disabled when CPU/Memory Request is larger than Limit